src/ElasticSearchRepository.php

```diff
@@ -44,34 +45,35 @@ public function __construct(
         Client $client,
         Serializer $serializer,
         string $index,
-        string $class,
+        string $class_type,
         array $notAnalyzedFields = []
     ) {
         $this->client = $client;
         $this->serializer = $serializer;
         $this->index = $index;
-        $this->class = $class;
+        $this->class_type = $class_type;
         $this->notAnalyzedFields = $notAnalyzedFields;
     }
 
     /**
-     * {@inheritdoc}
+     * @param Identifiable $data
+     * @throws MultiTypeIndexNotAllowedException
      */
     public function save(Identifiable $data): void
     {
-        Assertion::isInstanceOf($data, $this->class);
+        if (!$data instanceof $this->class_type) {
+            throw new MultiTypeIndexNotAllowedException(
+                "Data object should be of type {$this->class_type}, as declared on the repository definition."
+            );
+        }
 
         $serializedReadModel = $this->serializer->serialize($data);
-
-        $params = [
+        $this->client->index([
```

Comment:
So, giving it a step back and using a custom exception, how does it look like now? phpstan is happy with it